scripts/lib/CIME/tests/test_utils.py

```diff
@@ -1,15 +1,17 @@
 #!/usr/bin/env python3
 
 import os
+import shutil
 import sys
 import tempfile
 
 import unittest
 from unittest import mock
 from CIME.utils import indent_string, run_and_log_case_status, \
-    import_from_file
+    import_from_file, \
+    _line_defines_python_function, file_contains_python_function
```

Comment:
should we remove the underscore in _line_defines_python_function since we are calling it from outside CIME.utils?

Reply:
My thinking was that, since this should only be accessed by unit tests and not any external production code, I would keep it with the underscore. But if you feel I should remove the leading underscore due to its access by the unit tests, I'm happy to do so.

Reply:
No, fine as written.